src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -1,29 +1,28 @@
 """
 Contains all parts of pyTorch based machine learning model.
 """
-from typing import TYPE_CHECKING, Callable, Optional, Tuple, Union
+import logging
+from typing import Callable, Optional, Tuple
 
 import numpy as np
+from numpy.typing import NDArray
 
 from ...utils import maybe_progress
-from ..types import TwiceDifferentiable
+from .twice_differentiable import TwiceDifferentiable
 
 try:
     import torch
     import torch.nn as nn
     from torch import autograd
     from torch.autograd import Variable
 
-    _TORCH_INSTALLED = True
 except ImportError:
-    _TORCH_INSTALLED = False
-
-if TYPE_CHECKING:
-    from numpy.typing import NDArray
+    pass
```

Comment:
🤔 How can this work? If the code below relies on torch symbols imported here, if the import fails, the interpreter will choke. Have you tested in an environment without torch? If you want to make the stuff here available from `__init__.py`, the checking must be done there. Maybe try/except importing this module, and if you succeed, append the necessary symbols to `__all__` of that module.